src/Servers/Kestrel/Core/src/Middleware/HttpsConnectionMiddleware.cs

```diff
@@ -91,9 +93,18 @@ public HttpsConnectionMiddleware(ConnectionDelegate next, HttpsConnectionAdapter
             {
                 EnsureCertificateIsAllowedForServerAuth(_serverCertificate);
 
+                var certificate = _serverCertificate;
+                if (!certificate.HasPrivateKey)
```

Comment:
I thought that the plan was an explicit opt-in was needed before this compat path would be hit?

Reply:
Yea I discussed it with @halter73 and he felt strongly about not breaking anyone if we didn't have to.

Reply:
I think it make sense for Kestrel since that does not need to expose public API.